### PR TITLE
fx(params): add support for params in sandbox and mint generative token flow

### PR DESCRIPTION
**Pull request by maerzhase**
Long awaited PR that adds the support for fx(params) to the sandbox as well as the mint generative token support.

You can use this zip to test it. Its the params-boilerplate project ;-)

[project.zip](https://github.com/fxhash/fxhash-website/files/10871659/project.zip)

### CODE REVIEW

File: src/components/Artwork/SandboxPreview.tsx [hunk 1]
```diff
@@ -76,12 +77,12 @@ export const SandboxPreview = forwardRef<ArtworkIframeRef, Props>(
     // the URL of the iframe gets updated whenever ID / hash changes
     useEffect(() => {
       if (iframeRef.current && id !== "0") {
-        const previewUrl = `${location.origin}/sandbox/preview.html?id=${id}&fxhash=${hash}`
+        const previewUrl = `${location.origin}/sandbox/preview.html?id=${id}&fxhash=${hash}&fxparams=${fxparams}`
```
Comment (louisholley) [inline src/components/Artwork/SandboxPreview.tsx:80]: what about if it's not a params project? guessing this URL still works fine?

File: src/containers/MintGenerative/StepCheckFiles.tsx [hunk 2]
```diff
@@ -143,7 +172,7 @@ export const StepCheckFiles: StepComponent = ({ onNext, state }) => {
       <div className={cs(style.checkboxes)}>
         <div>
           <Checkbox value={check1} paddingLeft={false} onChange={setCheck1}>
-            I want to keep this hash for the preview of my project
+            I want to keep this settings for the preview of my project
```
Comment (louisholley) [inline src/containers/MintGenerative/StepCheckFiles.tsx:175]:
```suggestion
            I want to keep these settings for the preview of my project
```

File: src/containers/MintGenerative/StepCheckFiles.tsx [hunk 3]
```diff
@@ -26,58 +27,80 @@ export const StepCheckFiles: StepComponent = ({ onNext, state }) => {
...
+          } else {
+            setFeatures(null)
+          }
+        }
+        if (e.data.id === "fxhash_getParams") {
+          if (e.data.data) {
+            setParams(e.data.data)
+          } else {
+            setParams(null)
+          }
```
Comment (louisholley) [inline src/containers/MintGenerative/StepCheckFiles.tsx:65]: opinionated suggestion - easier to read with less nesting. this whole block could be

```
if (!e.data) return
if (e.data.id === "fxhash_getFeatures") setFeatures(e.data.data || null)
if (e.data.id === "fxhash_getParams") setParams(e.data.data || null)
```

File: src/containers/MintGenerative/StepExtraSettings/StepExtraSettings.tsx [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+      cloned.exploration?.postMint?.hashConstraints
+    ) {
+      cloned.exploration.postMint.hashConstraints = null
+    }
+    if (
+      postExploreOptions === "infinite" &&
+      cloned.exploration?.postMint?.paramsConstraints
+    ) {
+      cloned.exploration.postMint.paramsConstraints = null
+    }
```
Comment (louisholley) [inline src/containers/MintGenerative/StepExtraSettings/StepExtraSettings.tsx:182]: we could have something like

```
if (preExploreOptions === "infinite") {
  const preMint = cloned.exploration?.preMint
  const postMint = clone.exploration?.postMint
  if (preMint?.hashConstraints) cloned.exploration.preMint.hashConstraints = null
  ...
}
```

again, opinionated one, but always easier to understand/debug if there's less branching for me!

**Review by louisholley — APPROVED** [review]
awesome work man! few opinionated suggestion but feel free to ignore, lgtm 👍

File: src/components/Artwork/SandboxPreview.tsx (hunk 1 above)
Comment (maerzhase) [reply src/components/Artwork/SandboxPreview.tsx]: 👍

File: src/containers/MintGenerative/StepCheckFiles.tsx (hunk 3 above)
Comment (maerzhase) [reply src/containers/MintGenerative/StepCheckFiles.tsx]: good catch 👍

File: src/containers/MintGenerative/StepExtraSettings/StepExtraSettings.tsx (hunk 4 above)
Comment (maerzhase) [reply src/containers/MintGenerative/StepExtraSettings/StepExtraSettings.tsx:168]: jeah good point! thanks man!

**Pull request merged**